utils/writeable/src/cmp.rs

```diff
@@ -45,6 +46,51 @@ impl<'a> WriteComparator<'a> {
     }
 }
 
+/// Compares the contents of a `Writeable` to the given bytes
+/// without allocating a String to hold the `Writeable` contents.
+///
+/// This returns a lexicographical comparison, the same as if the Writeable
+/// were first converted to a String and then compared with `Ord`. For a
+/// locale-sensitive string ordering, use an ICU4X Collator.
```

Comment:
writeable should be independent of ICU4X, not sure we should recommend this here.

Reply:
+1

Reply:
This is pre-existing text.

Reply:
Yeah, that works.

Reply:
Not super happy with this, but won't block the PR as this is preexisting. `impl Ord for str`, and `str` I guess don't go into locale-aware collation either, and I consider this crate an alternative to those core types.